openverse_catalog/dags/providers/provider_api_scripts/science_museum.py

```diff
@@ -173,60 +172,51 @@ def _get_creator_info(obj_attr):
     return creator_info
 
 
-def _get_filesize(image_url):
-    """
-    Get the size of the image in bytes.
-    """
-    resp = delayed_requester.get(image_url)
-    if resp:
-        filesize = int(resp.headers.get("Content-Length", 0))
-        return filesize if filesize != 0 else None
-
-
-def _get_image_info(processed):
+def _get_image_info(
+    processed: Dict,
+) -> Tuple[str | None, int | None, int | None, str | None]:
+    height, width, filetype = None, None, None
     image_data = processed.get("large")
     if image_data is None:
         image_data = processed.get("medium", {})
 
-    image = image_data.get("location")
-    measurements = image_data.get("measurements")
-    filetype = image_data.get("format")
-    image = check_url(image)
-    height, width = _get_dimensions(measurements)
-    filesize = _get_filesize(image)
-    return image, height, width, filetype, filesize
+    image_url = check_url(image_data.get("location"))
+    if image_url:
+        filetype = image_data.get("format")
+        height, width = _get_dimensions(image_data)
+    return image_url, height, width, filetype
 
 
-def check_url(image_url):
-    base_url = "https://coimages.sciencemuseumgroup.org.uk/images/"
-    if image_url:
-        if "http" in image_url:
-            checked_url = image_url
-        else:
-            checked_url = base_url + image_url
+def check_url(image_url: str | None) -> str | None:
+    if not image_url:
+        return None
+    if image_url.startswith("http"):
+        return image_url
     else:
-        checked_url = None
-    return checked_url
+        return f"https://coimages.sciencemuseumgroup.org.uk/images/{image_url}"
 
 
-def _get_dimensions(measurements):
-    height_width = {}
-    if measurements:
-        dimensions = measurements.get("dimensions")
-        if dimensions:
-            for dim in dimensions:
-                height_width[dim.get("dimension")] = dim.get("value")
-    return height_width.get("height"), height_width.get("width")
+def _get_dimensions(image_data: Dict) -> Tuple[int | None, int | None]:
+    """
+    Returns the height and width of the image from a list of dictionaries
+    with keys of "dimension", "units", "value".
```

Comment:
This is not super important here, but we could use a [`TypedDict`](https://peps.python.org/pep-0589/) for cases like this! Also, it seems like it has a `measurements` key with a `dimensions` key inside the dictionary there?

Reply:
Thank you for the link, it looks interesting! Can we leave it for the next refactor 😉

I've changed the signature of this function. Instead of trying to get the `measurements` in one of two places using `measurements = processed.get("large / medium").get("measurements")`, I call the `_get_dimensions` with `image_data` we get using ``measurements = processed.get("large / medium")`, and then extract the measurements inside this function:
`height, width = _get_dimensions(image_data)`

Reply:
Yes, another time on that one! Just wanted to mention it :) 

And that works great! But I think the documentation should still read `with keys of "measurement"."dimensions", "units", and "value".` (since the top level key is still `measurements` rather than `dimensions`).